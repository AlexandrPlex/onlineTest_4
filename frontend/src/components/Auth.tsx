import * as hash from 'object-hash';
import * as React from 'react';

export interface IAuthProps {
  authState: boolean;
}

export interface IAuthDispatch {
    login: (email: string, password: string) => Promise<any>;
}

export class Auth extends React.Component<IAuthProps & IAuthDispatch> {
    public state = {
        emailValue: '',
        passwordValue: '',
    };
    public onChangeInput = (e: React.FormEvent<HTMLInputElement>) => {
        const state: any = {...this.state};
        state[e.currentTarget.name] = e.currentTarget.name === 'passwordValue' ? hash(e.currentTarget.value) : e.currentTarget.value;
        this.setState(state);
    }
    public onHandleSubmit = () => {
        alert(this.props.login(this.state.emailValue, this.state.passwordValue));
    }
    public render() {
        return (
            <React.Fragment>
                <form onSubmit={ (e: React.FormEvent<HTMLFormElement>) => { e.preventDefault(); } }>
                    <input onChange={this.onChangeInput} type='text' name='loginValue' placeholder='user name' />
                    <input onChange={this.onChangeInput} type='password' name='passwordValue' placeholder='password' />
                    <input onClick={this.onHandleSubmit} type='submit' name='submit'/>
                </form>
            </React.Fragment>
        );
    }
}
