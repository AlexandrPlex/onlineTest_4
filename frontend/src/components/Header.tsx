import * as React from 'react';

export interface IHeaderProps {
  authState: boolean;
}

export class Header extends React.Component<IHeaderProps> {
  public render() {
    return (
      <header>
        <nav>
        <span>Online tester</span>
          <ul>
            <li><a href='#'>Главная</a></li>
            {
              this.props.authState ?
              <li><a href='#'>Выйти</a></li>
              :
              <li><a href='#'>Войти</a></li>
            }
          </ul>
        </nav>
      </header>
    );
  }
}
