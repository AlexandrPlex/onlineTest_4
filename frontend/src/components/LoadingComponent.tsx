import * as React from 'react';

export class LoadingComponent extends React.PureComponent {
  public render() {
    return (
      <React.Fragment>
          <h1>LOADING</h1>
      </React.Fragment>
    );
  }
}
