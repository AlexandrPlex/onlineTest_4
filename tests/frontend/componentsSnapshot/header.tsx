import {Header, IHeaderProps} from 'components/Header';
import React from 'react';
import renderer from 'react-test-renderer';

describe('[SnapShot] Header test', () => {
    it('Auth true', ( ) => {
        const state: IHeaderProps = {
            authState: true,
        };
        const tree = renderer.create( <Header { ...state }/>);
        expect(tree).toMatchSnapshot();
    });
    it('Auth false', ( ) => {
        const state: IHeaderProps = {
            authState: false,
        };
        const tree = renderer.create( <Header { ...state }/>);
        expect(tree).toMatchSnapshot();
    });
});
