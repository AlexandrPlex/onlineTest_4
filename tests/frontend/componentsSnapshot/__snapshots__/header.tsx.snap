// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SnapShot] Header test Auth false 1`] = `
<header>
  <nav>
    <span>
      Online tester
    </span>
    <ul>
      <li>
        <a
          href="#"
        >
          Главная
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Войти
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`[SnapShot] Header test Auth true 1`] = `
<header>
  <nav>
    <span>
      Online tester
    </span>
    <ul>
      <li>
        <a
          href="#"
        >
          Главная
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          Выйти
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
